components/msg/constellation_msg.rs

```diff
@@ -458,8 +469,9 @@ pub fn start_monitoring_component(
     let component_id = MonitoredComponentId(pipeline, component_type);
     let bhm_chan = BackgroundHangMonitorChan::new(sender, component_id.clone());
     let thread_id = unsafe {
+        #[cfg(any(target_os = "android", target_os = "linux"))]
```

Comment:
A lot of this code would be nicer to modify outside of the msg crate. We might be able to define something like:
```rust
trait BackgroundHangMonitor {
    type ThreadId: Into<usize>;
    fn get_thread_id() -> Self::ThreadId;
    fn start_monitoring_component(...);
}
```
and move the majority of the implementation into a crate that only has components/servo depend on it (and include the BackgroundHangMonitor trait in the generics for ScriptThread and LayoutThread).

Reply:
Good idea, let me see what I can do...